homeassistant/config_entries.py

```diff
@@ -75,6 +75,14 @@
 SAVE_DELAY = 1
 
 
+class BaseServiceInfo:
+    """Base ServiceInfo type for discovery."""
+
+    def __getitem__(self, name: str) -> Any:
+        """Allow property access by name."""
+        return getattr(self, name)
```

Comment:
I don't think we want both ways to access the data since it will make it more difficult to review and we won't always get the benefits of using a dataclass.

I'd suggest implementing this as part of `DhcpServiceInfo` instead and logging a warning that this will fail in version 2022.6 when its accessed via `__getitem__` to use `<cls>.<name>` instead.